src/arcsf/models/trainer.py

```diff
@@ -0,0 +1,54 @@
+from datasets import Dataset
+from peft import PeftModel
+from transformers import (
+    DataCollatorForLanguageModeling,
+    EarlyStoppingCallback,
+    PreTrainedModel,
+    PreTrainedTokenizer,
+    Trainer,
+    TrainingArguments,
+)
+
+
+def load_trainer(
+    model: PreTrainedModel | PeftModel,
+    tokenizer: PreTrainedTokenizer,
+    dataset: Dataset,
+    trainer_kwargs: dict,
+    use_wandb: bool,
+) -> Trainer:
+
+    # Load training arguments
+    training_args = TrainingArguments(
+        **trainer_kwargs,
+        overwrite_output_dir=True,
+        report_to="wandb" if use_wandb else None,
```

Comment:
This behaviour's going to change in the next major version of `transformers` it seems, but setting `report_to=None` actually translates to "report to everything". `report_to="none"` or `report_to=[]` should do what's intended here.

Reply:
Ah good spot, thank you